src/ContMechTensors.jl

```diff
@@ -161,23 +161,23 @@ function call{dim}(Tt::Type{Vec{dim}}, data)
     Tensor{1, dim}(data)
 end
 
-## These are some kinda ugly stuff to create different type of constructors.
+# These are some kinda ugly stuff to create different type of constructors.
 @gen_code function call{order, dim}(Tt::Union{Type{Tensor{order, dim}}, Type{SymmetricTensor{order, dim}}},
                                           data)
     # Check for valid orders
-    if !(order in (1,2,4))
-        @code (throw(ArgumentError("Only tensors of order 1, 2, 4 supported")))
+    if get_type(Tt) <: Tensor
```

Comment:
Should probably just split this up into two functions now, one for Tensors and one for SymmetricTensors. Can do it later.